lib/portlayer/exec/exec.go

```diff
@@ -152,8 +158,36 @@ func eventCallback(ie events.Event) {
 
 			}
 		}
+		return
 	}
+	// check container registered event if this container is not found in container cache
+	switch ie.String() {
+	case events.ContainerRegistered:
+		moref := new(types.ManagedObjectReference)
+		if ok := moref.FromString(ie.Reference()); !ok {
+			log.Errorf("Failed to get event VM mob reference")
+			return
+		}
+		var vm mo.VirtualMachine
+
+		// current attributes we care about
+		attrib := []string{"config", "runtime.powerState", "summary", "resourcePool"}
 
+		// populate the vm properties
+		ctx := context.Background()
+		if err := eventSession.RetrieveOne(ctx, *moref, attrib, &vm); err != nil {
+			log.Errorf("Failed to query registered vm object %s: %s", ie.Reference(), err)
+			return
+		}
+		if *vm.ResourcePool != Config.ResourcePool.Reference() {
+			log.Debugf("vm %q is not containers of this VCH, ignore it", vm.Config.Name)
```

Comment:
Minor: I would reword this error to say "`container vm %q does not belong to this VCH, ignoring`" or something.